src/main/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysis.java

```diff
@@ -282,6 +410,36 @@ protected SecurityAnalysisResult runSimulations(LfNetwork lfNetwork, List<Propag
 
                                 postContingencyResults.add(postContingencyResult);
                                 networkState.restore();
+
```

Comment:
There is a lot of code in common between the breaking connectivity case and the non breaking connectivity case.
This code duplication will cause maintainance issues because modifications have to be reported to both branches of code.
What is the real difference between the breaking and non breaking connectivity case ? Is it only the call to calculatePostContingencyStatesForAContingencyBreakingConnectivity instead of calculatePostContingencyAndOperatorStrategyStatesForAContingency ?
In that case can case can you factorise the processing of a contingency (breaking connectivyity or not) in a method that would take an argument breaksConnectivity ?
That would make the code easier to understand and to maintain moving forward.

Reply:
Thanks. I think it is time to refactor because this PR shows that both branch have to evolve in parallel.